music_assistant/server/controllers/music.py

```diff
@@ -360,6 +360,14 @@ async def add_item_to_favorites(
             full_item.item_id,
             True,
         )
+        # ensure item is added to provider library
+        for provider_mapping in full_item.provider_mappings:
+            prov_controller = self.mass.get_provider(provider_mapping.provider_instance)
+            if prov_controller and prov_controller.library_edit_supported(full_item.media_type):
+                with suppress(NotImplementedError):
```

Comment:
Do not silently suppress here, that leads to hidden bugs if a (new) provider forgets to implement it.
Use a check if the provider is a streaming provider and has library edit feature.